Extending/Packages/Good-practice/index.md

```diff
@@ -0,0 +1,161 @@
+
+## License files
+Umbraco products store their licenses in `/umbraco/Licences` and it is considered a good location for third party packages that require license files to also store their licence files there. 
+
+_The default .gitignore for Umbraco templates will include any files in this folder in a repo (while ignoring most of the rest of the Umbraco folder)_
+
+:::note
+The `/umbraco/Licenses` folder does not exist on a fresh installation of Umbraco so you might need to create it before you save your license file to this folder. 
+:::
+
+## Operating system considerations
+Umbraco 9 is a .NET Core application and it can be ran on multiple operating systems (e.g Windows, Linux). When developing packages there are a few things you should be aware of so your package can run on all possible operating systems. 
+
+### Case sensitivity 
+The Linux file system is case sensitive, so `App_Plugins/myPackage` is a different location to `app_plugins/MyPACKAGE`. When building your package you should ensure that you always refer to folders and paths in a consistent way. 
+
+:::tip
+You can adjust the case sensitivity of a Windows folder by running a command against the folder. 
+
+``` bash
+fsutil.exe file queryCaseSensitiveInfo <path>
+```
+:::
+
+### Well known folders and case sensitivity  
+Some folders within Umbraco will already exist for all installations, if you access these folders you need to be aware of the case used to ensure you end up in the right place: 
+
+| Folder | Note | 
+|-|-|
+| /App_Plugins | Uppercase `A` and `P` |
+| /App_Plugins/[Ll]ang | Uppercase `L` but in later version of Umbraco 9.3+ can be upper or lowercase `L` |
+| /Views | Uppercase `V` | 
+| /umbraco/Licenses | Lowercase `u` uppercase `L` |
+| /config | Lowercase `c` |
+
+:::note 
+If you create a custom section/tree Umbraco will also build paths based on the name of that section or tree, these folder paths will also be case sensitive: 
+
+For example if you have a custom tree with the treeAlias of `MyCustomTree` Umbraco will look for files in `App_Plugins\MyCustomTree\backoffice\MyCustomTree\` 
+:::
+
+### File/folder locations
+You should never hardwire a file or folder location into code if possible. Instead you should either access files file the `IHostingEnvrionment` FileProviders or by using the built in methods to access well known locations 
+
+#### Temp folder
+The location of the Umbraco temp folder can be controlled via configuration and cannot be assumed, you should use the `IHostingEnvironment.LocalTempPath` variable to locate the temp folder. 
+
+``` cs
+var localTempRoot = Path.GetFullPath(hostingEnvironment.LocalTempPath);
+```
+
+#### Relative paths
+If you require the path of a folder relative to the site root, you can use the `IHostingEnvironment` methods to map a path. 
+
+``` cs 
+var contentRootPath = hostingEnvironment.MapPathContentRoot("/");
+var webrootPath = hostEnvironment.MapPathWebRoot("/");
```

Comment:
This should also show how you can create an absolute path to use as URL (and maybe also explain virtual paths/application relative paths):
```suggestion
var contentRootPath = hostingEnvironment.MapPathContentRoot("~/"); // Full physical path to the content/project root
var webrootPath = hostEnvironment.MapPathWebRoot("~/"); // Full physical path to the web root (served publicly)
var absolutePath = hostEnvironment.ToAbsolute("~/"); // Absolute path to use as URL
```